app/controllers/veterinary_offices_controller.rb

```diff
@@ -64,4 +37,46 @@ def destroy
   def vet_office_params
     params.permit(:name, :max_patient_capacity, :boarding_services)
   end
+
+  def veterinarians_filter
+    @veterinary_office = VeterinaryOffice.find(params[:veterinary_office_id])
+    if params[:sort] == "alphabetical"
```

Comment:
Instead of abstracting all the content from the controller action into another method, which still keeps the complexity all together, you could consider breaking up this method like so (also notice that this isn't a RESTful controller design, and could be handled in a different nested controller)

```
def veterinarians
       @veterinary_office = VeterinaryOffice.find(params[:veterinary_office_id])
       @office_vets = handle_params(params, @vetinerary_office)
end

... private

def handle_params(params, vet_office)
    if params[:sort] == "alphabetical"
      return vet_office.alphabetical_vets
    elsif params[:review_rating]
      return vet_office.office_vets_filtered_by_rating(
        params[:review_rating]
      )
    else
      return vet_office.on_call_vets
    end
end
```

Reply:
I prefer this refactoring rather than just having the call to the filter in the controller action because with this type of structure, you're not just abstracting *everything*, so it's easier to see what instance variables will be available in that view. It is encapsulating just the more complex piece (the if/elsif/else statement) but keeping the standard skeleton for the controller action.

Reply:
When you are saying this isn't a RESTful controller design, are you referring to the fact you are redirecting to vet_offices?

Reply:
I'm mostly referring to the fact that the application routes to a non-CRUD action in the controller, which is not the most RESTful routing (I'd opt for a nested vets controller under vet_offices)